src/nodes.coffee

```diff
@@ -258,6 +264,19 @@ exports.Base = class Base
     for child in @icedFlattenChildren()
       child.icedWalkAst o
 
+  # Traverse children and check if there is any await statement. This
+  # is used to generate an error when user tries to use an expression
+  # with Await in illegal places, like If's condition, Call arguments,
+  # right side of Assign, etc.
+  icedStatementAssertion : () ->
+    # TODO: Ideally this would be detected and saved in one of the
+    # tree-walking routines, so we don't walk the tree (potentially
+    # visiting same nodes over and over) every time someone calls this
+    # method.
```

Comment:
There is the `@icedFlag` which we set during our initial AST walk.  The slight issue is that we clear it as we process nodes. But we can work around that.